File: Java/Armor.java
import java.util.Random;

public class Armor {
	public double armorRating;
	public String type;
	public String name;

	public Random randomizer;
	
	public Armor(String Name, double ArmorPoints, String Type) {
		name = Name;
		type = Type;
		armorRating = ArmorPoints;
		}
	public Armor(String Name) {
		randomizer = new Random();
		name = Name;
		type = Name;
		if (type == "Helmet" ){
			armorRating = (double) randomizer.nextInt(50);
		}
		if(type == "Chestplate") {
			armorRating = (double) randomizer.nextInt(100);	
		}
		
		if(type == "Leggings") {
			armorRating = (double) randomizer.nextInt(75);	
		}
		
		if(type == "Boots") {
			armorRating = (double) randomizer.nextInt(50);	
		}
	}
	
	
	public Armor() {
		
		randomizer = new Random();
		int rantype = randomizer.nextInt(4);
		
		if (rantype == 0 ){
			armorRating = (double) randomizer.nextInt(50);
			type = "Helmet";
			name = type;
		}
		if(rantype == 1) {
			armorRating = (double) randomizer.nextInt(100);	
			type = "Chestplate";
			name = type;
		}
			
		if(rantype == 2) {
			armorRating = (double) randomizer.nextInt(75);	
			type = "Leggings";
			name = type;
		}
			
		if(rantype == 3) {
			armorRating = (double) randomizer.nextInt(50);
			type = "Boots";
			name = type;
		}	
	}
		
	public double getArmorRating() {
		return armorRating;	
	}
	
	public String getType() {
		return this.type;
	}
}

File: JavaScript/Armor.js

class Armor {
		
	constructor(type, level) {
		
		if (type == 0){
			this.armorRating = Math.floor(Math.random() * 15) + (15 * level);
			this.type = "Helmet";
		}
		if(type == 1) {
			this.armorRating = Math.floor(Math.random() * 40) + (15 * level);	
			this.type = "Chestplate";			
		}
			
		if(type == 2) {
			this.armorRating = Math.floor(Math.random() * 35) + (15 * level);
			this.type = "Leggings";			
		}
			
		if(type == 3) {
			this.armorRating = Math.floor(Math.random() * 15) + (15 * level);
			this.type = "Boots"			
		}	
	}
		
	getArmorRating() {
		return armorRating;	
	}
	
	getType() {
		return this.type;   
	}

    setBaseStats(){
        if(this.type == "Helmet"){
            this.armorRating = 15;
        }

        if(this.type == "Chestplate"){
            this.armorRating = 25;

        }
        if(this.type == "Leggings"){
            this.armorRating = 15;

        }
        if(this.type == "Boots"){
            this.armorRating = 15;
        }
        
    }
}

